aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -261,52 +262,88 @@ void printTensorNDArray(const Tensor& t) {
   [tdata printNDArray];
 }
 
-MPSCachedGraph* _getCachedGraph(const at::Tensor& src) {
+MPSCachedGraph* _getCachedGraph(const at::Tensor& src, GatherScatterViewOpType viewOpType) {
   MPSGraphCache* cache_ = MPSGraphCache::getInstance();
-  string key = getStridedKey(src, src.sizes(), src.strides(), src.storage_offset());
+  string key = getStridedKey(src, src.sizes(), src.strides(), src.storage_offset(), viewOpType);
   MPSCachedGraph* cachedGraph = cache_->LookUp(key);
 
   return cachedGraph;
 }
 
-id<MTLBuffer> _gatherViewTensor(const at::Tensor& src, id<MTLBuffer> sourceBuffer, MPSCachedGraph* mpsCachedGraph, Tensor& output) {
+id<MTLBuffer> _gattherScatterViewTensor(
+  const at::Tensor& src,
+  id<MTLBuffer> sourceBuffer,
+  MPSCachedGraph* mpsCachedGraph,
+  at::Tensor& output,
+  GatherScatterViewOpType viewOpType)
+{
   TORCH_CHECK(mpsCachedGraph != nil);
-
   MPSStream* stream = getCurrentMPSStream();
-
+  bool scatterViewOp = (viewOpType == GatherScatterViewOpType::Scatter);
   struct CachedGraph : public MPSCachedGraph
   {
     CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
     MPSGraphTensor* inputTensor_ = nil;
     MPSGraphTensor* outputTensor_ = nil;
+
+    MPSGraphTensor* scatteredTensor_ = nil;
+    MPSGraphTensor* updatesTensor_ = nil;
   };
```

Comment:
It's a bad practice to use the same inlined name over and over again and expect it to be the same. 
Perhaps we can move this definition to OperationUtils.mm and call it "IntOutScatterCachedGraph" or something?